### PR TITLE
Add support for prefix and suffix options

### PR DESCRIPTION
**Pull request by dwillett**
Adds support for `prefix` and `suffix` options similar to the options exposed for `store`

### CODE REVIEW

**Comment by dwillett** [comment]
I'm realizing that this approach doesn't really work since it actually changes the attribute name passed to `store_accessor`. I think I was right in the description, that it would need to pass the options through. Will take another pass at it.

**Comment by dwillett** [comment]
@byroot This should be ready for review now (with a revised approach) when you get a chance

File: lib/active_record/typed_store/dsl.rb [hunk 1]
```diff
@@ -6,10 +6,34 @@ module ActiveRecord::TypedStore
...
+      @coder = options.fetch(:coder) { default_coder(store_name) }
+      @store_name = store_name
+      @prefix =
+        case options[:prefix]
+        when String, Symbol
+          "#{options[:prefix]}_"
+        when TrueClass
+          "#{store_name}_"
+        else
+          ""
```
Comment (casperisfine) [inline lib/active_record/typed_store/dsl.rb:19]: I'd rather raise an `ArgumentError` if `prefix:` is an unexpected type.

File: lib/active_record/typed_store/dsl.rb [hunk 2]
```diff
@@ -6,10 +6,34 @@ module ActiveRecord::TypedStore
...
-      @accessors = options[:accessors]
-      @accessors = [] if options[:accessors] == false
+    def initialize(store_name, options)
+      @coder = options.fetch(:coder) { default_coder(store_name) }
+      @store_name = store_name
+      @prefix =
+        case options[:prefix]
+        when String, Symbol
+          "#{options[:prefix]}_"
+        when TrueClass
```
Comment (casperisfine) [inline lib/active_record/typed_store/dsl.rb:16]:
```suggestion
        when true
```

File: lib/active_record/typed_store/dsl.rb [hunk 3]
```diff
@@ -6,10 +6,34 @@ module ActiveRecord::TypedStore
...
+          "#{store_name}_"
+        else
+          ""
+        end
+      @suffix =
+        case options[:suffix]
+        when String, Symbol
+          "_#{options[:suffix]}"
+        when TrueClass
```
Comment (casperisfine) [inline lib/active_record/typed_store/dsl.rb:25]:
```suggestion
        when true
```

File: lib/active_record/typed_store/dsl.rb [hunk 4]
```diff
@@ -6,10 +6,34 @@ module ActiveRecord::TypedStore
...
+          "_#{store_name}"
+        else
```
Comment (casperisfine) [inline lib/active_record/typed_store/dsl.rb:27]: Same here. Let's raise if an unexpected type is passed.

File: lib/active_record/typed_store/dsl.rb [hunk 5]
```diff
@@ -6,10 +6,34 @@ module ActiveRecord::TypedStore
...
+          ""
+        end
+      @accessors = if options[:accessors] == false
+        {}
+      elsif options[:accessors].is_a?(Array)
```
Comment (casperisfine) [inline lib/active_record/typed_store/dsl.rb:32]: Is this another new feature?

Comment (dwillett) [reply lib/active_record/typed_store/dsl.rb:32]: This should have the same broad functionality for `@accessors` (can pass false to disable or nil to respect the field-specific accessor options) but accounts for the change in result type of `#accessors` to be a hash of `name => key` now

**Comment by casperisfine** [comment]
Arf, looks like CI is busted -_-

**Comment by casperisfine** [comment]
> Arf, looks like CI is busted -_-

I fixed it on master, you'll have to rebase though.

**Comment by dwillett** [comment]
Thanks - rebased and should be ready for another CI check

**Comment by byroot** [comment]
Can you please squash all your commits into one? After than I can merge.

**Comment by dwillett** [comment]
Sure thing, squashed and ready

**Comment by casperisfine** [comment]
Thanks!

**Comment by stellar-geoff** [comment]
👋  @casperisfine @byroot @dwillett Thanks for this PR, its just what I was looking for. It looks like this was merged but hasn't been included in a published version. Any chance you could publish a new one to rubygems? Thanks again either way

**Pull request merged**